### PR TITLE
Move pre-commit for local-file-mounts to breeze shell 

### PR DESCRIPTION
**Pull request by potiuk**
The pre-commit is moved to be run via breeze shell. The code to do
so has been migrated to new breeze command.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Based on #35875

File: BREEZE.rst [hunk 1]
```diff
@@ -1818,6 +1819,17 @@ nicely presented in ``--help`` output. You can check that via ``check-all-params
   :width: 100%
   :alt: Breeze setup check-all-params-in-group
 
+Breeze synchronize-local-mounts
+...............................
+
+When you add volumes mounted to docker, they need to be added in ``docker_command_utils.py`` - so that they
+are added by plain ``docker`` command, but they also nee to be synchronized with ``local.yml``. This can be
```
Comment (amoghrajesh) [inline BREEZE.rst]: typo: nee -> need.

File: scripts/ci/pre_commit/pre_commit_local_yml_mounts.py [hunk 2]
```diff
@@ -17,44 +17,18 @@
...
-                PREFIX + f"  target: {dest}\n",
-            ]
-        )
-    insert_documentation(local_mount_file_path, volumes, MOUNTS_HEADER, MOUNTS_FOOTER)
+res = subprocess.run(
+    ["breeze", "setup", "synchronize-local-mounts"],
+    check=False,
+)
+if res.returncode != 0:
+    console.print("\n[red]Error when running synchronization.\n")
```
Comment (amoghrajesh) [inline scripts/ci/pre_commit/pre_commit_local_yml_mounts.py]: nit: Error while running local mount synchronisation

**Review by amoghrajesh — COMMENTED** [review]
Generally looks good, small nits

**Comment by potiuk** [comment]
Same here another https://github.com/apache/airflow/pull/35830 off-spring

**Review by amoghrajesh — APPROVED** [review]

**Pull request merged**